ic-agent/src/agent/mod.rs

```diff
@@ -877,6 +878,19 @@ impl Agent {
             .map_err(|_| AgentError::CertificateVerificationFailed())
     }
 
+    fn verify_cert_timestamp(&self, cert: &Certificate) -> Result<(), AgentError> {
+        let time = lookup_time(cert)?;
+        if (OffsetDateTime::now_utc()
+            - OffsetDateTime::from_unix_timestamp_nanos(time.into()).unwrap())
+        .abs()
```

Comment:
Maybe this should check for timestamps that are in the future, but calling abs() and returning CertificateOutdated doesn't seem correct.

Reply:
That sounds like incorrect behavior on the part of agent-js then.  We shouldn't copy it here.

Reply:
See PERMITTED_DRIFT for how this is handled elsewhere https://github.com/dfinity/ic/blob/master/rs/constants/src/lib.rs#L15

Reply:
AIUI `PERMITTED_DRIFT` is for internal replica drift, where one subnet drifts from the other, and that this doesn't account for the caller's drift; is this accurate?

Reply:
I do not know the answer to that.

Reply:
Well, for now I'll just delete the abs check then. If you're that far in the past, the replica would have already rejected your expiry timeout as being in the past.